swarm/src/behaviour.rs

```diff
@@ -122,7 +119,7 @@ pub(crate) type THandlerOutEvent<THandler> =
 ///   }
 /// }
 /// ```
-pub trait NetworkBehaviour: 'static {
+pub trait NetworkBehaviour: Sized + 'static {
```

Comment:
This requires a `Sized` bound now but I'd honestly not know why a `NetworkBehaviour`wouldn't be sized ...

Reply:
How is the requirement to be `Sized` related to this change?

Reply:
I think it is because I am now using `Self` within a function signature via a type-alias. Frankly, I just followed the compiler :upside_down_face:

Reply:
I would prefer to avoid casually introducing trait bounds without careful consideration. Is there any way around introducing this trait bound? Is this pull request worth doing given the added complexity.

I would have to dig deeper here. Based on intuition, adding `Sized`, `Send` or `Sync` unintentionally has not proven to be good ideas in the past.

Reply:
Fair point! I never looked deeply into it but apparently, requiring `Sized` prohibits the use as a trait object: https://doc.rust-lang.org/std/marker/trait.Sized.html

I'll try to work around it.

Reply:
Turns out it is not needed, lol :joy:

Reply:
> Fair point! I never looked deeply into it but apparently, requiring `Sized` prohibits the use as a trait object:
> 
> I'll try to work around it.

yeah, that's why you can't have `Box<dyn Clone>` for example

Reply:
Thanks for investigating. Great to see it not being needed.